### PR TITLE
SaveDashboard: Display error message when folder/dashboard name validation fails

### LINKED ISSUE #22966: Folder Creation does not check for Dashboard name conflict
**What happened**:
I created a folder with the same name as an existing dashboard. Then went to edit the dashboard. When I tried to save the dashboard I got an error message that a dashboard cannot be the same name as a folder.

**What you expected to happen**:
I would expected that if this were the case. Then I shouldn't have been allowed to create the folder, as there was an existing dashboard.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a Dashboard with a given name
2. Create a folder with the same name
3. Attempt to edit and save existing dashboard
4. Get error regarding folder/dashboard name conflict

**Anything else we need to know?**:
This error occurs on 6.6.2. However, when I upgraded to 6.7.1 I no longer get an error, but the dashboard still does not save correctly.

**Environment**:
- Grafana version: 6.6.2/6.7.1
- Data source type & version: N/A
- OS Grafana is installed on: Alpine 3.11.3 (running in a Docker Container though)
- User OS & Browser: Windows 10 + Firefox 75.0b7
- Grafana plugins: grafana-worldmap-panel, grafana-clock-panel, grafana-piechart-panel, ryantxu-ajax-panel, farski-blendstat-panel, agenty-flowcharting-panel, jdbranham-diagram-panel, michaeldmoore-multistat-panel, natel-plotly-panel, grafana-polystat-panel, snuids-radar-panel, pr0ps-trac
- Others:

### PR DESCRIPTION
**Pull request by dprokop**
Fixes #22966

### CODE REVIEW

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx [hunk 1]
```diff
@@ -67,15 +65,30 @@ export const SaveDashboardAsForm: React.FC<SaveDashboardFormProps & { isNew?: bo
...
-            <Input name="title" ref={register({ required: true })} aria-label="Save dashboard title field" autoFocus />
+          <Forms.Field label="Dashboard name" invalid={!!errors.title} error={errors.title?.message}>
+            <Input
+              name="title"
+              ref={register({
+                validate: v => {
+                  if (v.trim() === '') {
+                    return 'Dashboard name is required';
+                  } else if (v && v === getValues().$folder.title?.trim()) {
+                    return 'Dashboard name cannot be the same as folder';
```
Comment (tskarhed) [inline public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: I never reach this line. This is what I get when logging `getValues` 
Screenshot from 2020-04-08 10-38-36

File: public/app/features/folders/components/NewDashboardsFolder.tsx [hunk 2]
```diff
@@ -37,8 +37,8 @@ export class NewDashboardsFolder extends PureComponent<Props> {
       .then(() => {
         return true;
       })
-      .catch(() => {
-        return 'Folder already exists.';
+      .catch(e => {
+        return e.message;
```
Comment (tskarhed) [inline public/app/features/folders/components/NewDashboardsFolder.tsx:41]: Really like this improvement! :)
Screenshot from 2020-04-08 10-23-23

**Review by tskarhed — CHANGES_REQUESTED** [review]

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: weirdo, will have a look. What exactly did you do?

Comment (tskarhed) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: The error appeared when creating a dashboard in the `General` folder with the same name as a top level folder.

Comment (tskarhed) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: I did reach this line when I created a dashboard with the same name as a folder. Shoudln't that be possible though?
Screenshot from 2020-04-08 10-42-03

File: public/app/features/folders/components/NewDashboardsFolder.tsx (hunk 2 above)
Comment (dprokop) [reply public/app/features/folders/components/NewDashboardsFolder.tsx:41]: this is not possible. See the referenced issue

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: this is not possible. See the referenced issue

Comment (dprokop) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: @tskarhed I've tried to reproduce but whad do you mean by "Top level folder"? General is a top level folder. We don't have nested folders structure. So I cannot make the folder title to be undefined.

Comment (Clarity-89) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx:84]: The validation error above will be triggered only when creating a dashboard with the same name as its folder. However, the backend call will also fail if a dashboard is created with the same name as any other existing folder. 
So this catches the error:
Screenshot 2020-04-09 at 10 20 27
While this doesn't:
Screenshot 2020-04-09 at 10 22 05

Comment (dprokop) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx]: @tskarhed this should be fixed now. Do you mind reviewing again?

**Review by tskarhed — APPROVED** [review]
Looks good, awesome work!

**Pull request merged**